src/actions/deviceActions.js

```diff
@@ -473,7 +473,7 @@ export function setPowerMode(isSmuMode) {
         await device.ppkSetPowerMode(isSmuMode ? 2 : 1);
         logger.info(`Mode: ${isSmuMode ? 'Source meter' : 'Ampere meter'}`);
         dispatch(setPowerModeAction(isSmuMode));
-        if (!isSmuMode) dispatch(setDeviceRunning(true));
+        dispatch(setDeviceRunning(!isSmuMode));
```

Comment:
In case we actually use this:

When we switch from ampere to source mode, we first switch the mode and then turn the power off. So I assume there is a short moment when we are in source mode but power is turned on. To avoid this, I suggest:
1. If we are switching from source to ampere mode, then first switch the mode and then the power on.
2. If we are switching from ampere to source mode, then first turn the power off and then switch the mode.

And, BTW, let us move that `isSmuMode ? 2 : 1` to `ppkSetPowerMode` in `serialDevice.js`. That is technical detail that does not belong here.

Reply:
Sharp eye there 👍